packages/input-select/src/Select.js

```diff
@@ -0,0 +1,159 @@
+import * as React from "react";
+import ReactAutocomplete from "react-autocomplete";
+import PropTypes from "prop-types";
+import { compose } from "recompose";
+import disabledTooltip, {
+  disabledTooltipProps
+} from "@crave/farmblocks-hoc-disabled-tooltip";
+import formInput, { formInputProps } from "@crave/farmblocks-hoc-input";
+import {
+  DropdownWrapper,
+  DropdownMenuWrapper,
+  DropdownItemWrapper
+} from "@crave/farmblocks-dropdown";
+
+import EmptyCard from "./components/EmptyCard";
+
+const EnhancedInput = compose(disabledTooltip, formInput)("input");
+EnhancedInput.displayName = "EnhancedInput";
+
+class Select extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedValue: props.value,
+      selectedLabel: this.getSelectedLabel(props),
+      isSearching: false,
+      isMenuOpen: false
+    };
+
+    this._renderInput = this._renderInput.bind(this);
+    this._renderMenu = this._renderMenu.bind(this);
+    this._renderItem = this._renderItem.bind(this);
+    this._shouldItemRender = this._shouldItemRender.bind(this);
+    this.onFilter = this.onFilter.bind(this);
+    this.onSelect = this.onSelect.bind(this);
+  }
+
+  render() {
+    return (
+      <DropdownWrapper>
+        <ReactAutocomplete
+          items={this.props.items}
+          shouldItemRender={this._shouldItemRender}
+          getItemValue={item => item.label}
+          value={this.state.selectedLabel}
+          onChange={this.onFilter}
+          onSelect={this.onSelect}
+          renderInput={this._renderInput}
+          renderMenu={this._renderMenu}
+          renderItem={this._renderItem}
+          autoHighlight={false}
+          onMenuVisibilityChange={isMenuOpen => this.setState({ isMenuOpen })}
+        />
+      </DropdownWrapper>
+    );
+  }
+
+  onFilter(event) {
+    !this.state.isSearching && this.props.onChange("");
+
+    this.setState({ selectedLabel: event.target.value, isSearching: true });
+  }
+
+  onSelect(selectedLabel, item) {
+    this.setState({ selectedLabel, isSearching: false });
+    this.props.onChange(item.value);
+  }
+
+  getSelectedLabel(props) {
+    const item = props.value && props.items.find(x => x.value === props.value);
+
+    return item && item.label;
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (
+      newProps.value &&
+      newProps.value !== this.props.value &&
+      newProps.value !== this.state.selectedValue
+    ) {
+      this.setState({
+        selectedValue: newProps.value,
+        selectedLabel: this.getSelectedLabel(newProps)
+      });
+    }
+  }
+
+  _renderInput(autoCompleteProps) {
+    const { ref, ...rest } = autoCompleteProps;
+
+    const { renderItem, ...inputProps } = this.props;
+
+    inputProps.validationMessages = this.state.isMenuOpen
+      ? []
+      : this.props.validationMessages;
+
+    return <EnhancedInput {...inputProps} {...rest} innerRef={ref} />;
```

Comment:
we should have and option to render a select box where the input is not a text input IMHO, just like a regular html select box... but we can handle this on a separated enhancement issue

Reply:
Agreed

Reply:
filed #243